lib/elixir/lib/record.ex

```diff
@@ -36,6 +36,10 @@ defmodule Record do
 
   """
 
+  @type type :: :defrecord | :defrecordp
+  @type name :: atom
+  @type tag :: atom
```

Comment:
`name` is used in `Record.extract/2`. I think it's important to have a distinction between name and tag as both are atoms, but it wasn't clear in the Record module whether we are dealing with the name or the tag.